src/Microsoft.Azure.SignalR.Common/IServiceEndpointGenerator.cs

```diff
@@ -1,12 +1,14 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Microsoft.AspNetCore.Http;
+
 namespace Microsoft.Azure.SignalR
 {
     internal interface IServiceEndpointGenerator
     {
         string GetClientAudience(string hubName);
-        string GetClientEndpoint(string hubName, string originalPath);
+        string GetClientEndpoint(string hubName, string originalPath, QueryString queryString);
```

Comment:
Thinking about moving it back to `Microsoft.Azure.SignalR` project, as this interface won't be the same as `Microsoft.Azure.SignalR.AspNet`.
/cc @vicancy

Reply:
`Microsoft.Azure.SignalR.AspNet` is not using it. But I am thinking if I should use it so that we can aware such change for `Microsoft.Azure.SignalR.AspNet`, is `queryString` also needed for `Microsoft.Azure.SignalR.AspNet`?

Reply:
I don't think you will need it. Client SDK will preserve the original query string when getting redirect response.